docs/ref/models/fields.txt

```diff
@@ -190,6 +190,14 @@ scenes.
 
 If ``True``, a database index will be created for this field.
 
+.. note::
+
+    When using the MySQL database backend, a database index won't be created
```

Comment:
I don't know if documenting this list is a good solution given the "errors should not pass silently" philosophy. Also, some users are probably note aware of the underlying database types that different fields use. I think if third-party apps weren't a concern, there would be no problem having the system check framework raise an error.  Perhaps a system check warning would be appropriate. Then users can silence that if the lack of an index isn't a problem for their use case.